server/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -95,6 +95,13 @@ public final void collectExistingBucket(LeafBucketCollector subCollector, int do
         subCollector.collect(doc, bucketOrd);
     }
 
+    /**
+     * Merges the bucket doc counts together using <code>mergeMap</code> where <code>mergeMap[i]</code> is the
+     * ordinal of the bucket that the i'th bucket is merged into
+     *
+     * This only tidies up doc counts. Call <code>DeferringCollector::mergeBuckets</code> to merge the actual
```

Comment:
If you use `{@link DeferringCollection#mergeBuckets}` then javadoc will check that the method on the other side exists.